OasisDEX/oasis-borrow
# Added BasicBS readonly flag and handled edge cases

# [Added BasicBS readonly flag and handled edge cases](https://app.shortcut.com/oazo-apps/story/5112/separate-out-feature-flags-to-read-and-write)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- added feature flag `ReadOnlyBasicBS` which when enabled doesn't allow for adding new basic buy or sell triggers but allows for removing them
- added handling for cases where vault has 0 debt and enabled basic buy / sell triggers
- added handling for optimization tab when vault has 0 debt (banner)
- updated copies
- minor fixes

## How to test 🧪
  <Please explain how to test your changes>

- when vault has 0 debt and basic buy / sell triggers enabled user should be able to remove them
- when vault has 0 debt it should be impossible to add new triggers
- when vault has debt and `ReadOnlyBasicBS` and `BasicBS` is enabled user shouldn't be able to add new triggers but if he has already basic sell or buy, should be able to remove them
